ast/impl/array_length_enforcer.cpp

```diff
@@ -0,0 +1,103 @@
+#include "visitors.hpp"
+
+namespace visitors {
+
+using namespace ast;
+
+void ArrayLengthEnforcer::visit(Program* node) {
+    for (auto var : node->variables) {
+        var->type->accept(*this);
+    }
+    for (auto routine : node->routines) {
+        routine->accept(*this);
+    }
+}
+
+void ArrayLengthEnforcer::visit(RoutineDecl* node) {
+    insideParameters = true;
+    for (auto param : node->parameters) {
+        param->type->accept(*this);
+    }
+    insideParameters = false;
```

Comment:
Shouldn't there be a `m_` prefix for name of global variable?

Reply:
It is not a global variable, it is a private variable inside this class only.
Also, I didn't really understand what `m_` and `g_` mean

Reply:
g_ stands for global.
m_ I don't know. But is looks better that a leading or trailing underscore.